src/model.cpp

```diff
@@ -274,6 +274,18 @@ bool Model::hasUnlinkedUnits()
     return unlinkedUnits;
 }
 
+bool Model::isDefined() const
+{
+    bool defined = true;
+    for (size_t n = 0; n < unitsCount() && defined; ++n) {
```

Comment:
This is forcing all units to be defined, including units that may not be used. Historically we have normally only cared that the units that are actually used are defined (i.e., just to below component loop). Might be interesting to run the "loop over PMR and flatten all models" script to see if this could be an actual problem...

Reply:
I took the approach of the validation here, where we look at the whole model/document.